### ISSUE TITLE
Unable to sync certain folders

### ISSUE THREAD

**Issue by highorbit25** [Beebeeoii/lominus#111]
**Describe the bug**
Certain folders within a module cannot be synced

**To Reproduce**
1. Sync for CS2102
2. Only Folders 'Cases', 'course_image' synced
3. Missing Folders 'Finals', 'Guide', 'Midterms', 'Slides', 'Tutorials'
'Finals' and 'Midterms' Folder are locked and inaccessible on Canvas so non-issue

**Expected behavior**
'Guide', 'Slides', 'Tutorials' Folder should be able to sync

**Screenshots**
Canvas: 
![image](https://github.com/Beebeeoii/lominus/assets/74848536/d2bedff3-23ab-4f23-980e-4dc2a9b0be5b)
Synced by Lominus:
![image](https://github.com/Beebeeoii/lominus/assets/74848536/accccf01-112a-4497-97ea-1901401a28dc)
Logs:
`[2024-01-29 01:13:42] WARNING - json: cannot unmarshal object into Go value of type []interfaces.CanvasFileObject (C:/Program Files/lominus-development_290124/internal/cron/cron.go:github.com/beebeeoii/lominus/internal/cron.createJob.func1:162)

[2024-01-29 01:13:43] WARNING - json: cannot unmarshal object into Go value of type []interfaces.CanvasFileObject (C:/Program Files/lominus-development_290124/internal/cron/cron.go:github.com/beebeeoii/lominus/internal/cron.createJob.func1:162)

[2024-01-29 01:13:43] WARNING - json: cannot unmarshal object into Go value of type []interfaces.CanvasFileObject (C:/Program Files/lominus-development_290124/internal/cron/cron.go:github.com/beebeeoii/lominus/internal/cron.createJob.func1:162)`

**Desktop (please complete the following information):**
 - OS: Microsoft Windows 10 Home
 - Version: latest dev branch 290124, not working on v2.1.0 as well

**Additional context**
Issue might be from unsupported folder/file configurations on Canvas.
Observed within tutorials folder, tutorial 'T01.pdf' has a scheduled release configuration
![image](https://github.com/Beebeeoii/lominus/assets/74848536/9e9a0c66-e94b-476e-a833-a93f4e168c06)

Let me know if more clarifications needed! Thanks 👍

**Comment by Beebeeoii** [comment]
Hi, thanks for the report.

Could you check for me if:
1) the folders that are missing ("guides", "slides", "tutorials") actually have files inside? Lominus will not create those folders if there are no files inside. In other words, it will not create empty folders.
2) If there are files inside, those files are available for download normally through Canvas. Lominus cannot download files that are not released, locked etc.

Thanks!

**Comment by highorbit25** [comment]
Yes, the missing folders all have files inside, and can be downloaded normally through Canvas.
![image](https://github.com/Beebeeoii/lominus/assets/74848536/392fd046-0e5c-46d1-8562-4f056e4ee04a)
![image](https://github.com/Beebeeoii/lominus/assets/74848536/c2807ef6-9621-48b8-b8f6-accd18457238)
![image](https://github.com/Beebeeoii/lominus/assets/74848536/566573f3-465c-4f79-9cab-2f4f0dd81ad5)
Thanks!

**Comment by Beebeeoii** [comment]
Weird issue as this issue is because of Canvas returning an error 403 Forbidden even though user is able to access the files within the folder normally via the website.

https://github.com/instructure/canvas-lms/issues/2306
